### PR TITLE
[jest]: change default generic value to `{}`

### PR DESCRIPTION
**Pull request by G-Rath**
Addresses [comment in this PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44433#issuecomment-628789936)

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44433#issuecomment-628789936
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by tonyhallett — APPROVED** [review]

**Review by favna — CHANGES_REQUESTED** [review]
~~I don't really see any good reason to do this. Can you give a good reason for your changes?~~

~~Also~~ Please properly finish up the pull request template (remove the non-relevant entries and the text that reminds you to do so)

Edit: nevermind, I didn't see the link at the bottom because you didn't edit the template fully. Please do that and I'll give my approval.

Edit 2: For clarifications. Remove the squared blocks, move the reason above.
image

**Review by favna — APPROVED** [review]

**Review by r3nya — APPROVED** [review]

**Comment by alexandcote** [comment]
I tried to use the changes you proposed and this doesn't seem to solve the issue I raised here.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44433#issuecomment-628789936

I still get the error.

image

If I change the type for something like 
```
function requireActual<TModule extends {} = {}>(moduleName: string): TModule;
function requireMock<TModule extends {} = {}>(moduleName: string): TModule;
```

This solve the issue.

**Comment by tonyhallett** [comment]
image

It is due to the return type of the factory being unknown.

As per @alexandcote - the change below resolves the issue.

> function requireActual<TModule extends {} = {}>(moduleName: string): TModule;
function requireMock<TModule extends {} = {}>(moduleName: string): TModule;

File: types/jest/index.d.ts [hunk 1]
```diff
@@ -176,12 +176,12 @@ declare namespace jest {
      * Returns the actual module instead of a mock, bypassing all checks on
      * whether the module should receive a mock implementation or not.
      */
-    function requireActual<TModule = any>(moduleName: string): TModule;
+    function requireActual<TModule = {}>(moduleName: string): TModule;
```
Comment (tonyhallett) [inline types/jest/index.d.ts:179]: TModule extends {} = {}

File: types/jest/index.d.ts [hunk 2]
```diff
@@ -176,12 +176,12 @@ declare namespace jest {
...
     /**
      * Returns a mock module instead of the actual module, bypassing all checks
      * on whether the module should be required normally or not.
      */
-    function requireMock<TModule = any>(moduleName: string): TModule;
+    function requireMock<TModule = {}>(moduleName: string): TModule;
```
Comment (tonyhallett) [inline types/jest/index.d.ts:184]: TModule extends {} = {}

File: types/jest/jest-tests.ts [hunk 3]
```diff
@@ -286,15 +286,15 @@ jest.advanceTimersToNextTimer();
 jest.advanceTimersToNextTimer(2);
 
 // https://jestjs.io/docs/en/jest-object#jestrequireactualmodulename
-// $ExpectType any
+// $ExpectType {}
```
Comment (tonyhallett) [inline types/jest/jest-tests.ts:289]: Add
jest.mock('module',()=>({...jest.requireActual('module')}))

**Review by tonyhallett — CHANGES_REQUESTED** [review]
Perhaps then it is the factory return type that should be changed to any and requireActual and requireModule can remain as default any.

@ExE-Boss 
Ok to change unknown to any to facilitate this change ?
`
function mock(moduleName: string, factory?: () => unknown, options?: MockOptions): typeof jest;
`
[commit introduced](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/5ce5329b565d816a5a9506a5a8671ca758ecb307)

**Pull request closed without merging**